oracle/src/events/processCollectedSignatures/index.js

```diff
@@ -46,6 +49,37 @@ function processCollectedSignaturesBuilder(config) {
         logger.info(`Processing CollectedSignatures ${colSignature.transactionHash}`)
         const message = await homeBridge.methods.message(messageHash).call()
 
+        if (ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST || ORACLE_HOME_TO_FOREIGN_BLOCK_LIST) {
+          const parsedMessage = parseMessage(message)
+          const recipient = parsedMessage.recipient.toLowerCase()
+          const originalTxHash = parsedMessage.txHash
+
+          if (ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST) {
+            const allowanceList = await readAccessListFile(ORACLE_HOME_TO_FOREIGN_ALLOWANCE_LIST)
+            if (allowanceList.indexOf(recipient) === -1) {
+              const sender = (await web3Home.eth.getTransaction(originalTxHash)).from.toLowerCase()
```

Comment:
To get sender for large amount of transactions to check the address allowance and for almost every transaction to check the blocked address seems for me as a quite costly operation. Can we have additional option to enable this check, e.g. `ORACLE_HOME_TO_FOREIGN_CHECK_SENDER`? This parameter could be omitted in the configuration file so it is considered as `false`.